HDDS-3571. Recon: Display leader count in Datanodes page

vivekratnavel opened the pull request:
## What changes were proposed in this pull request?

- Update Node endpoint to include leader count and pipelines object to include leader node
- Make changes in the frontend to add a new column "leader count" in Datanodes page
- Update pipeline icon to show if the datanode is the leader of pipeline or not in Datanodes page
- Update unit tests to test leader count

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-3571

## How was this patch tested?

This patch was tested in docker-compose env with multi-raft enabled.
Add the following configs to docker-config to enable multi-raft.
OZONE-SITE.XML_ozone.datanode.pipeline.limit=5
OZONE-SITE.XML_ozone.scm.ratis.pipeline.limit=0

<img width="1680" alt="Screen Shot 2020-05-12 at 5 09 49 PM" src="https://user-images.githubusercontent.com/1051198/81879002-58e2f000-953e-11ea-9579-37dbb31b209a.png">

xiaoyuyao commented on hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/api/NodeEndpoint.java:
```diff
@@ -81,31 +83,44 @@ public Response getDatanodes() {
       String hostname = datanode.getHostName();
       Set<PipelineID> pipelineIDs = nodeManager.getPipelines(datanode);
       List<DatanodePipeline> pipelines = new ArrayList<>();
+      AtomicInteger leaderCount = new AtomicInteger();
```
I think this is to work around local variable in capturing lambdas must be final or effective final.

avijayanhwx replied:
Ah yes. I am OK with using AtomicInteger then.

vivekratnavel replied:
@xiaoyuyao That is correct. Since increment happens inside the lambda, a regular int cannot be used.

vivekratnavel commented:
@avijayanhwx @xiaoyuyao Thanks for the reviews. I have updated the patch to fix review comments and updated the db.json file to fix the mock API server to serve correct API responses.

avijayanhwx commented:
LGTM +1
